### DISCUSSION TITLE
Display progress bar %

### DISCUSSION THREAD

**Discussion by discodevl** [google/model-viewer#3521] [Q&A] [answered]
is there a way to show the grey progress bar percentage after uploading a glb?
![Untitled design](https://user-images.githubusercontent.com/69987663/173329170-b0beded8-397e-43a4-8177-38bbb27826c7.png)

**Comment by samaneh-kazemi** [comment, marked as answer]
Hi, you can override the default progress bar[using slot] and control/set the value to display using javascript.

Example of how to use progress bar:

```
<style>
.progress-bar {
      ...
    }
</style>
<model-viewer id="mv_id" camera-controls reveal="interaction" src="../../shared-assets/models/glTF-Sample-Models/2.0/DamagedHelmet/glTF/DamagedHelmet.gltf" alt="A 3D model of a damaged helmet">
  <div id="progress-bar-d" slot="progress-bar"></div>
</model-viewer>
<script>
// Set the percentage by querying progress-bar-id and setting the value to sth like percentage of the width
</script>
```

**Comment by elalish** [comment]
You might also be looking for our progress event, if you want to write the number as text: https://modelviewer.dev/docs/index.html#entrydocs-loading-events-progress
